apps/web/src/hooks/useCapabilitiesSafe.ts

```diff
@@ -0,0 +1,37 @@
+/*
+  A hook to safely check wallet_getCapabilities support
+
+  Responsabilities:
+  - Check for unsupported wallets (e.g: metamask)
+  - Use experimental useCapabilities
+  - Check atomicBatch (batchcall) and paymasterService for a given chain
+  - Default to false
+*/
+
+import { Chain } from 'viem';
+import { useAccount } from 'wagmi';
+import { useCapabilities } from 'wagmi/experimental';
+
+export type UseCapabilitiesSafeProps = {
+  chain: Chain;
+};
+
+export default function useCapabilitiesSafe({ chain }: UseCapabilitiesSafeProps) {
```

Comment:
can you add a comment about what makes this "safe"?

Reply:
it's in the comment above !